src/bom.rs

```diff
@@ -44,18 +46,31 @@ pub struct Bom<'a> {
     #[serde(serialize_with = "uuid_to_urn")]
     serial_number: Uuid,
     version: u32,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub metadata: Option<Metadata<'a>>,
     components: Vec<Component<'a>>,
 }
 
-/// Create a new BOM from a sequence of cargo package references.
-impl<'a> FromIterator<&'a Package> for Bom<'a> {
-    fn from_iter<T: IntoIterator<Item = &'a Package>>(iter: T) -> Self {
+impl<'a> Default for Bom<'a> {
+    fn default() -> Bom<'a> {
         Self {
             bom_format: BomFormat::CycloneDX,
             spec_version: SPEC_VERSION,
             version: 1,
             serial_number: Uuid::new_v4(),
-            components: iter.into_iter().map(Component::library).collect(),
+            metadata: None::<Metadata>,
+            components: vec![],
```

Comment:
For an empty vector, the macro produces this code anyway, so I tend to just write `Vec::new()` in these situations
```suggestion
            components: Vec::new(),
```

Reply:
In: 0940426